mmv1/templates/terraform/examples/artifact_registry_repository_remote_custom_dockerhub_auth.tf.erb

```diff
@@ -0,0 +1,42 @@
+data "google_project" "project" {}
+
+resource "google_secret_manager_secret" "<%= ctx[:vars]['secret_resource_id'] %>" {
+  secret_id = "<%= ctx[:vars]['secret_id'] %>"
+  replication {
+    auto {}
+  }
+}
+
+resource "google_secret_manager_secret_version" "<%= ctx[:vars]['secret_resource_id'] %>_version" {
+  secret = google_secret_manager_secret.<%= ctx[:vars]['secret_resource_id'] %>.id
+  secret_data = "<%= ctx[:vars]['secret_data'] %>"
+}
+
+resource "google_secret_manager_secret_iam_member" "secret-access" {
+  secret_id = google_secret_manager_secret.<%= ctx[:vars]['secret_resource_id'] %>.id
+  role      = "roles/secretmanager.secretAccessor"
+  member    = "serviceAccount:service-${data.google_project.project.number}@gcp-sa-artifactregistry.iam.gserviceaccount.com"
+}
+
+resource "google_artifact_registry_repository" "<%= ctx[:primary_resource_id] %>" {
+  location      = "us-central1"
+  repository_id = "<%= ctx[:vars]['repository_id'] %>"
+  description   = "<%= ctx[:vars]['description'] %>"
+  format        = "DOCKER"
+  mode          = "REMOTE_REPOSITORY"
+  remote_repository_config {
+    description = "custom uri docker hub with auth credentials"
+    disable_upstream_validation = true
+    docker_repository {
+      custom_repository {
+        uri = "<%= ctx[:vars]['custom_uri'] %>"
+      }
```

Comment:
@hao-nan-li this is the test that is failing. As you can see I'm only specifying `custom_repository`, but the test is failing because the `public_repository` default is also being applied and both can't be applied